File: src/main/java/com/netdetpla/ndp/bean/ResponseEnvelope.java
package com.netdetpla.ndp.bean;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class ResponseEnvelope <T> {
    private Integer code;
    private String message;
    private T data;

    public Integer getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }

    public ResponseEnvelope(Integer code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }
    public ResponseEnvelope(Integer code, String message) {
        this.code = code;
        this.message = message;
    }

}

File: src/main/kotlin/com/ndp/server/bean/ResponseEnvelope.kt
package com.ndp.server.bean

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_EMPTY)
class ResponseEnvelope<T> {
    var code: Int
        private set
    var message: String
        private set
    var data: T? = null
        private set

    constructor(code: Int, message: String, data: T) {
        this.code = code
        this.message = message
        this.data = data
    }

    constructor(code: Int, message: String) {
        this.code = code
        this.message = message
    }
}